### PR TITLE
Fixes #8 remove il code

### LINKED ISSUE #8: Remove IL code used in Delegate Factory in favor of ExpressionCompile()

### PR DESCRIPTION
**Pull request by msevestre**
This is removing some very old optimization code that should be done differently since version 4.5.x

### CODE REVIEW

**Comment by msevestre** [comment]
Also update .NET version to 4.5.2 since this is the default that we are installing everywhere and 4.5 is barely supported anymore

File: src/OSPSuite.Utility/Reflection/DelegateFactory.cs [hunk 1]
```diff
@@ -1,106 +1,85 @@
...
+    /// <summary>
+    ///     Delegate to a property get info.
+    ///     This is required when using reflection extensively to enhance performance
+    /// </summary>
+    public delegate void SetHandler(object target, object value);
+
+    public static class DelegateFactory
+    {
+        /// <summary>
+        ///     Returns an handler to a property get for a property that is compiled=>access is much faster than using reflection
```
Comment (Yuri05) [inline src/OSPSuite.Utility/Reflection/DelegateFactory.cs:21]: _a_ handler :)

File: src/OSPSuite.Utility/Reflection/DelegateFactory.cs [hunk 2]
```diff
@@ -1,106 +1,85 @@
...
+            var lambda = Expression.Lambda<GetHandler>(
+                Expression.Convert(member, typeof(object)),
+                instanceParameter
+            );
+
+            return lambda.Compile();
+        }
+
+        /// <summary>
+        ///     Returns an handler to a property get for a field that  is compiled=>access is much faster than using reflection
```
Comment (Yuri05) [inline src/OSPSuite.Utility/Reflection/DelegateFactory.cs:38]: dito

File: src/OSPSuite.Utility/Reflection/DelegateFactory.cs [hunk 3]
```diff
@@ -1,106 +1,85 @@
...
+            var lambda = Expression.Lambda<GetHandler>(
+                Expression.Convert(member, typeof(object)),
+                instanceParameter
+            );
+
+            return lambda.Compile();
+        }
+
+        /// <summary>
+        ///     Returns an handler to a property set for a field that is compiled=>access is much faster than using reflection
```
Comment (Yuri05) [inline src/OSPSuite.Utility/Reflection/DelegateFactory.cs:55]: same

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**